### PR TITLE
WT-6247 Apply WT_SESSION_RESOLVING_MODIFY to the update chain

### PR DESCRIPTION
**Pull request by quchenhao**
When cleaning WT-6246, I found this problem.

### CODE REVIEW

**Comment by quchenhao** [comment]
@tetsuo-cpp Since now we don't need to manipulate the read timestamp when reconstructing modify in history store, can we simplify things in the txn structure? I remember you and Michael did something there but I don't know what exactly.

**Comment by tetsuo-cpp** [comment]
> @tetsuo-cpp Since now we don't need to manipulate the read timestamp when reconstructing modify in history store, can we simplify things in the txn structure? I remember you and Michael did something there but I don't know what exactly.

@quchenhao Yeah I think we can simplify this now. I'll push a commit shortly if that's ok.

**Comment by tetsuo-cpp** [comment]
@quchenhao 
Just pushed a change to remove the mutable `read_timestamp` from `WT_TXN` since we no longer move it back and forth when processing modifies. My commit bloats your PR by quite a bit so if you don't like this, feel free to revert and I can follow up with another PR to consolidate.

**Comment by quchenhao** [comment]
@tetsuo-cpp That's a lot of change, I think we should move that to a separate pr.

**Comment by tetsuo-cpp** [comment]
@quchenhao 
Understood. I've just reverted and will open another PR.

File: src/include/txn.i [hunk 1]
```diff
@@ -931,7 +938,7 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_ITEM *key, uint
     }
 
     /* If the start time point is visible then we need to return the ondisk value. */
-    if (__wt_txn_tw_start_visible(session, &tw) || F_ISSET(session, WT_SESSION_RESOLVING_MODIFY)) {
+    if (F_ISSET(session, WT_SESSION_RESOLVING_MODIFY) || __wt_txn_tw_start_visible(session, &tw)) {
```
Comment (tetsuo-cpp) [inline src/include/txn.i:941]: We just swapped the conditions?

**Comment by quchenhao** [comment]
Can someone look at this?

File: src/include/txn.i [hunk 2]
```diff
@@ -736,6 +736,13 @@ __wt_txn_upd_visible_type(WT_SESSION_IMPL *session, WT_UPDATE *upd)
         if (prepare_state == WT_PREPARE_LOCKED)
             continue;
 
+        if (F_ISSET(session, WT_SESSION_RESOLVING_MODIFY) && upd->txnid != WT_TXN_ABORTED &&
```
Comment (tetsuo-cpp) [inline src/include/txn.i:739]: What about consecutive modifies?

Comment (quchenhao) [reply src/include/txn.i:739]: What do you mean? It should work the same.

File: src/include/txn.i (hunk 1 above)
Comment (quchenhao) [reply src/include/txn.i:941]: The first check is more lightweight so I think we should swap the sequence.

File: src/include/txn.i (hunk 2 above)
Comment (tetsuo-cpp) [reply src/include/txn.i:739]: Sorry, I'm probably misunderstanding. I just didn't get why we're checking specifically for a standard update here.

Comment (quchenhao) [reply src/include/txn.i:739]: We don't want to ignore history store tombstone here.

Comment (quchenhao) [reply src/include/txn.i:739]: Otherwise we still get nothing

Comment (tetsuo-cpp) [reply src/include/txn.i:739]: Oops, sorry I got confused. The modify is going to be written as a standard update in the history store.

**Comment by tetsuo-cpp** [comment]
@luke-pearson @kommiharibabu 
Can we get one more approval? I have a PR pointing at this branch and I'd like to get it pointing at `develop` asap.

**Pull request merged**